packages/driver/src/cy/commands/navigation.ts

```diff
@@ -342,23 +378,77 @@ const stabilityChanged = (Cypress, state, config, stable) => {
 
     debug('waiting for window:load')
 
-    return new Promise((resolve) => {
-      return cy.once('window:load', (e) => {
+    const promise = new Promise((resolve) => {
+      const onWindowLoad = (win) => {
         // this prevents a log occurring when we navigate to about:blank inbetween tests
         if (!state('duringUserTestExecution')) return
 
         cy.state('onPageLoadErr', null)
 
-        if (e.window.location.href === 'about:blank') {
+        if (win.location.href === 'about:blank') {
           // we treat this as a system log since navigating to about:blank must have been caused by Cypress
           options._log.set({ message: '', name: 'Clear Page', type: 'system' }).snapshot().end()
         } else {
           options._log.set('message', '--page loaded--').snapshot().end()
         }
 
-        return resolve()
+        resolve()
+      }
+
+      const onCrossOriginWindowLoad = ({ url }) => {
+        options._log.set('message', '--page loaded--').snapshot().end()
+
+        // Updating the URL state, This is done to display the new url event when we return to the primary origin
+        let urls = state('urls') || []
+        let urlPosition = state('urlPosition')
+
+        if (urlPosition === undefined) {
+          urlPosition = -1
+        }
+
+        urls.push(url)
+        urlPosition = urlPosition + 1
+
+        state('urls', urls)
+        state('url', url)
+        state('urlPosition', urlPosition)
+
+        resolve()
+      }
+
+      const onCrossOriginFailure = (err) => {
+        options._log.set('message', '--page loaded--').snapshot().end()
+        options._log.set('state', 'failed')
+        options._log.set('error', err)
+
+        resolve()
+      }
+
+      const onInternalWindowLoad = (details) => {
+        switch (details.type) {
+          case 'same:origin':
```

Comment:
nit: move these constants to the top

Reply:
What would be the benefit? We don't use these anywhere else in this file, the string is as good as a const defined at the top of the file.